.github/workflows/tests.yml

```diff
@@ -6,32 +6,30 @@ on:
   pull_request:
 
 jobs:
-  Tests:
+  Core:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-18.04]
-
+        os: [macos-latest, ubuntu-20.04]
     steps:
       - uses: actions/checkout@v2
       - name: Run tests
-        run: .github/workflows/scripts/run_tests.sh
+        timeout-minutes: 30
+        run: .github/workflows/scripts/run_tests_core.sh
+      - name: CPP Lint
+        run: .github/workflows/scripts/lint_cpp.sh
+        if: ${{ matrix.os == 'ubuntu-20.04' }}
 
-  Linters:
-    runs-on: ubuntu-latest
+  Bazel:
+    runs-on: ubuntu-20.04
     steps:
-      - name: Install toolchains
+      - name: Install golang
         uses: actions/setup-go@v2
         with:
           go-version: 1.13.3
         id: go
-      - name: Install linters
+      - name: Install buildifier
         run: go get github.com/bazelbuild/buildtools/buildifier
       - uses: actions/checkout@v2
-      - name: C++ Build
-        run: .github/workflows/scripts/build_cpp.sh
```

Comment:
The build_cpp step was added for preventing warnings at build time, such as size_t vs int comparisons etc.

I didn't want to enforce that on every platform, that's why it was I separate step. 
Is SEAL generating warnings now on Ubuntu?

Reply:
@bcebere In the test workflow it doesn't show warnings. Is that what you mean?

Reply:
It was supposed to fail on warnings at build time, it had the -Werrror flag in addition to the regular build.

Reply:
After discussing internally, we've come to the conclusion it is safe to merge without the above build script.